rt/databinding/jaxb/src/main/java/org/apache/cxf/jaxb/FactoryClassProxyService.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.jaxb;
+
+import org.apache.cxf.Bus;
+
+public class FactoryClassProxyService implements FactoryClassCreator {
+    FactoryClassCreator srv;
+    public FactoryClassProxyService(Bus bus) {
+        this(new FactoryClassGenerator(bus));
+    }
+    public FactoryClassProxyService(FactoryClassCreator srv) {
+        super();
+        this.srv = srv;
+    }
+
+    @Override
+    public Class<?> createFactory(Class<?> cls) {
+        return srv.createFactory(cls);
+    }
+
+    public class LoadFirst extends FactoryClassProxyService {
+        public LoadFirst(ClassLoader cl) {
+            //TODO not sure here if I get class loader like that ???
```

Comment:
Is the comment still relevant? It seems like you the correct classloader

Reply:
loadFirst will be initialize by outside code I guess but if we change it in bus-extensions.txt it will failed because bus extensions expect a Bus as first argument. So I guess, I have to put bus for every loadFirst ctor. On master, we have the system with Map of class and classloader and I think it was done on purpose to avoid the following cases:
2 requests arrive => 2 threads are created (right ?) => each one want to get a class generated => get a classloader which are common on all thread (need synchronized) => ry to find class => if not found generate class and load them in classLoader. I think that s why class loader was on a static map to handle multi thread. wdyt ?

Reply:
I don't think so, the static variables were a lazy+leaky impl IMHO.
Generating twice the class is not an issue while the classloader ensures it does findClass before any registration in a correctly locked context.
Worse case you lost some CPU cycles but didn't block a thread.